### PR TITLE
Added custom pooling csv file for LCMB, tests

### LINKED ISSUE #1329: DPL-825 Provision a new csv file to be downloadable at the PCR-XP stage in the LCMB pipeline
**User story**
Provision a new concentration CSV file to be downloadable at the PCR-XP plate stage in the LCMB pipeline. See exports_controller for how these are done (see details of columns for csv below). This information will be used to determine the custom pooling strategy in the next step of the pipeline.

**Who are the primary contacts for this story**
Ben F. Liz E.

**Who is the nominated tester for UAT**
Ben F.

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] A new concentration CSV file to be downloadable at the PCR-XP stage

In Integration Suite:
- [x] Update LCMB test to include check for file download availability and file expected content, and confirm test runs successfully

**Additional context**
***Concentration CSV file details***
- barcode of source plate (Stock or Cherrypick)
- sample name
- well position
- concentration (nanograms/microlitre -  ng/ul)
- SS sample ID?
- "shotgun?" (extra column with empty value)
- "ISC?" (extra column with empty value)

**References**
This story has been split from #702

### PR DESCRIPTION
**Pull request by andrewsparkes**
Closes #1329 

Changes proposed in this pull request:

Added file export for concentrations file on XP plate

### CODE REVIEW

File: spec/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  let(:ancestor_well_a1) do
+    create(:v2_well, position: { 'name' => 'A1' }, qc_results: create_list(:qc_result, 1, qc_result_options_1))
+  end
+  let(:ancestor_well_b1) do
+    create(:v2_well, position: { 'name' => 'B1' }, qc_results: create_list(:qc_result, 1, qc_result_options_2))
+  end
+  let(:labware) { create(:v2_plate, wells: [well_a1, well_b1], pool_sizes: [1, 1]) }
+  let(:ancestor_labware) { create(:v2_plate, wells: [ancestor_well_a1, ancestor_well_b1], pool_sizes: [1, 1]) }
```
Comment (harrietc52) [inline spec/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb_spec.rb:25]: How/where is the labware associated with the ancestor labware?

**Review by harrietc52 — APPROVED** [review]
Looks fine :) just a few questions for my understanding

File: app/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<% @plate.wells_in_columns.each_with_index do |well, well_index| %>
+  <% unless well.empty? || @ancestor_plate.blank? %>
+    <% ancestor_well = @ancestor_plate.wells_in_columns[well_index] %>
+<%= CSV.generate_line [
+    @ancestor_plate.human_barcode,
+    well.aliquots.first.sample.name,
+    well.location,
+    well.latest_concentration&.value,
+    well.aliquots.first.sample.sanger_sample_id,
+    nil,
```
[the remark this answers is not in the document]
Comment (andrewsparkes) [reply app/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb:19]: Yes, the idea is for the user to download this file, make some decisions about next steps for each sample, and presumably flag those decisions in those columns in the file.
The next step is custom pooling, where they select a subset for sequencing.

File: spec/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb_spec.rb (hunk 1 above)
Comment (andrewsparkes) [reply spec/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb_spec.rb:25]: It isn't associated.
I think we bypass that normal requirement by assigning the plate and ancestor parameters for the screen in the before section lines 35-38. i.e. the screen doesn't care, and it's the screen we are testing here.

Comment (andrewsparkes) [reply spec/views/exports/lcmb_pcr_xp_concentrations_for_custom_pooling.csv.erb_spec.rb:25]: Could add a comment?
I'm also now wondering why the qc results are being assigned to both ancestor and current plate wells...
This was a partial cut/paste from another similar test.

**Pull request merged**